src/client/components/DiceInterface.tsx

```diff
@@ -228,13 +484,43 @@ export function DiceInterface() {
                         : bonuses.toString()}
                     </div>
                   </Button>
-                  <Button className="w-full" onClick={() => clear()}>
+                  <Button
+                    className="w-full"
+                    onClick={() => {
+                      clear();
+                      focusIndexFromRef(clearRef);
+                    }}
+                    ref={clearRef}
+                  >
                     Clear Input
                   </Button>
                 </td>
               </tr>
             </tbody>
           </table>
+          <p>
+            {previousRolls.map((roll) => {
+              const ref = prevRollRefs[previousRolls.indexOf(roll)];
+              return (
+                <Button
+                  key={roll.diceTypes.join("") + String(roll.bonuses)}
+                  onClick={async () => {
+                    await doRoll(false, roll);
+                    focusIndexFromRef(prevRollRefs[0]);
+                  }}
+                  ref={ref}
+                  style={{ width: "33.3%" }}
+                >
+                  {roll.diceTypes.join(" + ")}{" "}
+                  {roll.bonuses === null
+                    ? ""
+                    : roll.bonuses >= 0
+                    ? "+" + roll.bonuses.toString()
+                    : roll.bonuses.toString()}
```

Comment:
While we're at it, would you mind changing this and other occurrences of adding `"+"` manually to numbers with `signedModifierString(roll.bonuses)`?